frontend/src/metabase/nav/containers/MainNavbar/SidebarItems/DwhUploadCSV/DwhUploadCSV.tsx

```diff
@@ -0,0 +1,94 @@
+import { type ChangeEvent, useCallback, useRef, useState } from "react";
+import { t } from "ttag";
+
+import {
+  type CollectionOrTableIdProps,
+  ModelUploadModal,
+} from "metabase/collections/components/ModelUploadModal";
+import type { OnFileUpload } from "metabase/collections/types";
+import { UploadInput } from "metabase/components/upload";
+import { useToggle } from "metabase/hooks/use-toggle";
+import { useDispatch } from "metabase/lib/redux";
+import {
+  type UploadFileProps,
+  uploadFile as uploadFileAction,
+} from "metabase/redux/uploads";
+import { Box, Button, Icon } from "metabase/ui";
+
+import { trackDWHUploadCSVClicked } from "./analytics";
+
+export const DwhUploadCSV = () => {
+  const [
+    isModelUploadModalOpen,
+    { turnOn: openModelUploadModal, turnOff: closeModelUploadModal },
+  ] = useToggle(false);
+
+  const [uploadedFile, setUploadedFile] = useState<File | null>(null);
+
+  const dispatch = useDispatch();
+  const uploadFile = useCallback(
+    ({ file, modelId, collectionId, tableId, uploadMode }: UploadFileProps) =>
+      dispatch(
+        uploadFileAction({ file, modelId, collectionId, tableId, uploadMode }),
+      ),
+    [dispatch],
+  );
+
+  const handleFileUpload = useCallback<OnFileUpload>(
+    (uploadFileArgs: CollectionOrTableIdProps) => {
+      const { collectionId, tableId } = uploadFileArgs;
+      if (uploadedFile && (collectionId || tableId)) {
+        closeModelUploadModal();
+        uploadFile({
+          file: uploadedFile,
+          ...uploadFileArgs,
+        });
+      }
+    },
+    [uploadFile, uploadedFile, closeModelUploadModal],
+  );
+
+  const handleFileInput = (event: ChangeEvent<HTMLInputElement>) => {
+    trackDWHUploadCSVClicked();
+    const file = event.target.files?.[0];
+
+    if (file !== undefined) {
+      setUploadedFile(file);
+      openModelUploadModal();
+
+      // reset the input so that the same file can be uploaded again
+      if (uploadInputRef.current) {
+        uploadInputRef.current.value = "";
+      }
+    }
+  };
+
+  const uploadInputRef = useRef<HTMLInputElement>(null);
+  const triggerUploadInput = () => uploadInputRef.current?.click();
+
+  return (
+    <Box data-testid="dwh-upload-csv" mt="md" px="md">
+      <Button
+        color="brand"
+        fullWidth={true}
+        leftIcon={<Icon name="upload" />}
+        onClick={triggerUploadInput}
+        radius="xl"
+        variant="outline"
+      >
+        {t`Upload CSV`}
+      </Button>
```

Comment:
out of scope for this PR, I assume, but would it be hard to make this button be a file drop zone? I found myself wanting to drop a CSV file onto it.

Reply:
To help out the translator maybe add a little context:

```ts
{c('Text for a button that lets you upload a CSV file').t`Upload CSV`}
```

This makes it super-clear that "Upload" is a verb not an adjective (as in "put the upload CSV here, and the download CSV there").

Reply:
I think the Figma gives this button .5rem padding on top and bottom so it's ~32px tall like the other menu items

Reply:
> I think the Figma gives this button .5rem padding on top and bottom so it's ~32px tall like the other menu items

Yes, but we don't have 32px tall button in our design system. I will not introduce a custom component. The designer is aware of this and is ok with the button being the default size (40px).